pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -218,6 +225,10 @@ type Controller struct {
 
 	// Network name for the registry as specified by the MeshNetworks configmap
 	networkForRegistry string
+	// tracks which services on which ports should act as a gateway for networkForRegistry
+	registryServiceNameGateways map[host.Name]uint32
```

Comment:
Keeping in mind that we'll switch to BTS, and with BTS we may use POST-over-H2 - it may be better long term to use a URL instead of hostname and port. 

If you really want to allow the port to be customized in the gateway service - it opens the possibility that a user may attach multiple Services with different ports to a gateway name - which can be a legitimate use case to segment access in a cluster, or with external load balancers ( i.e. as a poor man substitute for BTS ).

It may be ok to fix it in a separate PR, but I'm afraid it'll make it messier to add BTS.

Reply:
The map here would go away soon, I hope. We use it to lookup a service on each service update. If it's in there, then it was included as part of the legacy meshNetworks registryServiceName config.